app/src/main/java/org/stacks/app/ui/secret/SecretKeyActivity.kt

```diff
@@ -0,0 +1,63 @@
+package org.stacks.app.ui.secret
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.android.synthetic.main.activity_secret_key.*
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import org.stacks.app.R
+import org.stacks.app.ui.BaseActivity
+import org.stacks.app.ui.secret.bottomsheet.ShareKeyBottomSheetFragment
+import reactivecircus.flowbinding.android.view.clicks
+
+@AndroidEntryPoint
+class SecretKeyActivity : BaseActivity() {
+
+    private lateinit var viewModel: SecretKeyViewModel
+
+    private val backButtonEnabled by lazy {
+        intent?.getBooleanExtra(BACK_BUTTON, false) ?: false
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_secret_key)
+        if (backButtonEnabled) {
+            setNavigation()
+        }
+
+        viewModel = ViewModelProvider(this).get(SecretKeyViewModel::class.java)
+
+        viewModel
+            .secretKey()
+            .onEach {
+                secretKey.text = it
+            }
+            .launchIn(lifecycleScope)
+
+        copyKeyButton
+            .clicks()
+            .map { viewModel.copyPressed() }
```

Comment:
Weird mapping out a viewModel call with no return value. You probably just wanted to call it, and could have done so inside the `onEach` below.